MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m

```diff
@@ -148,8 +189,100 @@ - (MSPriority)priority {
   return MSPriorityHigh;
 }
 
-- (MSInitializationPriority)initializationPriority {
-  return MSInitializationPriorityDefault;
+#pragma mark - Private
+
+- (BOOL)checkURLSchemeRegistered:(NSString *)urlScheme {
+  NSArray *schemes;
+  NSArray *types = [MS_APP_MAIN_BUNDLE objectForInfoDictionaryKey:@"CFBundleURLTypes"];
+  for (NSDictionary *urlType in types) {
+    schemes = urlType[@"CFBundleURLSchemes"];
+    for (NSString *scheme in schemes) {
+      if ([scheme isEqualToString:urlScheme]) {
+        return YES;
+      }
+    }
+  }
+  return NO;
+}
+
+- (NSURL *)buildTokenRequestURLWithAppSecret:(NSString *)appSecret error:(NSError *__autoreleasing *)error {
+
+  // Compute URL path string.
+  NSString *urlPath = [NSString stringWithFormat:kMSUpdtsUpdateTokenApiPathFormat, appSecret];
+
+  // Build URL string.
+  NSString *urlString = [kMSDefaultInstallUrl stringByAppendingString:urlPath];
+  NSURLComponents *components = [NSURLComponents componentsWithString:urlString];
+
+  // Check URL validity so far.
+  if (!components && error) {
+    NSString *desc = [NSString stringWithFormat:@"%@\n%@", kMSUDUpdateTokenURLInvalidErrorDesc, components];
+    *error = [NSError errorWithDomain:kMSUDErrorDomain
+                                 code:kMSUDUpdateTokenURLInvalidErrorCode
+                             userInfo:@{NSLocalizedDescriptionKey : desc}];
+    return nil;
+  }
+
+  // Set URL query parameters.
+
+  // FIXME: Workaround to fill in the app name required by the backend for now, supposed to be a build UUID.
+  NSString *buildUUID = [MS_APP_MAIN_BUNDLE objectForInfoDictionaryKey:@"MSAppName"];
+  //    NSString *buildUUID = [[MSFTCECodeSignatureExtractor forMainBundle]
+  //    getUUIDHashHexStringAndReturnError:&buildError];
+  //    if (buildError && error) {
+  //      *error = error;
+  //      return nil;
+  //    }
+  NSMutableArray *queryItems = [NSMutableArray array];
+  if (![self checkURLSchemeRegistered:kMSUpdtsDefaultCustomScheme] && error) {
```

Comment:
I would keep it there, like validating the query items when using them makes sens. Also moving it would not bring that much performances. If we would have been able to do all the validation at the same place I would have done it.